### ISSUE TITLE
docker build failed with golang-go.tools conflicts with golang-go

### ISSUE THREAD

**Issue by chengweiv5** [sameersbn/docker-gitlab#445]
I was trying to build docker image but it failed to run install.sh like below.

```
Preparing to unpack .../golang-go.tools_0.0~hg20131126-3_amd64.deb ...
Unpacking golang-go.tools (0.0~hg20131126-3) ...
dpkg: error processing archive /var/cache/apt/archives/golang-go.tools_0.0~hg20131126-3_amd64.deb (--unpack):
 trying to overwrite '/usr/lib/go/pkg/tool/linux_amd64/vet', which is also in package golang-go 2:1.5
dpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
```

Is there a way to workaround such this error?

**Comment by sameersbn** [comment]
@chengweiv5 I am not sure why you are seeing this error. have you done any changes?

**Comment by STOIE** [comment]
Yeah, I am seeing this as well.... fresh build from trusty and also seeing it in fresh build from utopic.

Been an open bug for several weeks now, this is a bummer since I really want to get the v8 build up and running.

https://bugs.launchpad.net/ubuntu/+source/golang-go.tools/+bug/1486564

```
Preparing to unpack .../make_3.81-8.2ubuntu3_amd64.deb ...
Unpacking make (3.81-8.2ubuntu3) ...
Selecting previously unselected package golang-go.tools.
Preparing to unpack .../golang-go.tools_0.0~hg20131126-3_amd64.deb ...
Unpacking golang-go.tools (0.0~hg20131126-3) ...
[91mdpkg: error processing archive /var/cache/apt/archives/golang-go.tools_0.0~hg20131126-3_amd64.deb (--unpack):
 trying to overwrite '/usr/lib/go/pkg/tool/linux_amd64/vet', which is also in package golang-go 2:1.5
[0m[91mdpkg-deb: error: subprocess paste was killed by signal (Broken pipe)
[0mSelecting previously unselected package libffi-dev:amd64.
Preparing to unpack .../libffi-dev_3.1~rc1+r3.0.13-12_amd64.deb ...
Unpacking libffi-dev:amd64 (3.1~rc1+r3.0.13-12) ...
Selecting previously unselected package libyaml-dev:amd64.
Preparing to unpack .../libyaml-dev_0.1.4-3ubuntu3.1_amd64.deb ...
Unpacking libyaml-dev:amd64 (0.1.4-3ubuntu3.1) ...
Selecting previously unselected package paxctl.
Preparing to unpack .../paxctl_0.7-1_amd64.deb ...
Unpacking paxctl (0.7-1) ...
Selecting previously unselected package ruby2.1-dev:amd64.
Preparing to unpack .../ruby2.1-dev_2.1.7-1bbox1~trusty1_amd64.deb ...
Unpacking ruby2.1-dev:amd64 (2.1.7-1bbox1~trusty1) ...
[91mErrors were encountered while processing:
 /var/cache/apt/archives/golang-go.tools_0.0~hg20131126-3_amd64.deb
[0m[91mE: [0m[91mSub-process /usr/bin/dpkg returned an error code (1)
[0mThe command '/bin/sh -c bash ${SETUP_DIR}/install.sh' returned a non-zero code: 100
Error response from daemon: could not find image: no such id: dd/gitlab:latest
Error response from daemon: could not find image: no such id: dd/gitlab:latest
The push refers to a repository [infra.ris.environment.gov.au/gitlab] (len: 1)
Error pushing to registry: Tag does not exist: 3a9e3fc7df6715790da645b9491338d3a5923904_33
The push refers to a repository [infra.ris.environment.gov.au/gitlab] (len: 1)
```

**Comment by STOIE** [comment]
+1

**Comment by chengweiv5** [comment]
@sameersbn Yes, I changed the base image to official ubuntu and replace apt-get mirrors to a local one, so it could be faster.

BTW, I could not reproduce on the vanilla code base. I think this bug introduced by changes I made, free free to close this one, thanks.

**Comment by sameersbn** [comment]
p.s. https://github.com/sameersbn/docker-gitlab/blob/master/assets/setup/install.sh#L8 adds the ppa for trusty. If you change the base image to anything other than trusty, then you would need to update the line appropriately.

**Comment by STOIE** [comment]
So my build uses down-stream official ubuntu:trusty, and then adds my proxy environment bits to it:
https://github.com/envris/corporate-proxy-base/blob/trusty/Dockerfile

and then gosu:
https://github.com/envris/docker-gitlab-govt/blob/master/Dockerfile

and lastly the buildout exactly as per this repo, using the previous two up-streams as from bases.

**Comment by sameersbn** [comment]
@STOIE are you able to build the image without any changes?

**Comment by STOIE** [comment]
I did a little review and noticed your upstream no longer includes gosu and has some other new bits.

So, I have now replicated that:
https://github.com/envris/docker-gitlab-govt/blob/master/Dockerfile

And it appears I now have a working build (still using my proxy base) which is awesome!

It hasn't fully built yet (gems seemingly take an eternity), but it's further than it was getting before.

Thanks again,
Aaron.
